### ISSUE TITLE
count() returns something like [ { '': 1 }]

### ISSUE THREAD

**Issue by ericbiewener** [tgriesser/knex#3315]
# Environment

Knex version: 0.18.0
Database + version: MS SQL 2017
OS: Mac 10.14.5

# Bug
The docs say that `.count()` should return `number | string`, but I'm getting `[ { '': 1 } ]`. My code is simply:

```js
const results = await db(TABLE_NAME).count()
console.log(results) // [ { '': 1 } ]
```

Perhaps this is a MS SQL specific issue?

**Comment by lorefnon** [comment]
I have found this very confusing too but the docs don't really say that. 

Knex just runs a `count(*)` operation and maps the resulting single row table to an array of objects just like any other query result. So the key name you end up with in that dictionary is specific to the database you are using. 

So you'd want to do something like: 
```
(await db(TABLE_NAME)
    .count('count' /* used as the column name */  ))
    .count
``` 

I'd love to see this change but it would be breaking.

**Comment by kibertoad** [comment]
@lorefnon Can't we introdu a differently names count() method and start showing deprecation warning for old one?

**Comment by lorefnon** [comment]
Do you have a good name in mind ? 😅

**Comment by kibertoad** [comment]
@lorefnon Most obvious would be `countAsNumber` and `countAsString`, including appropriate conversions when necessary. Alternatively: `countReduced`, `countFlat`.

**Comment by elhigu** [comment]
I believe that docs are broken at this time and .count() actually should return a row with count, like DB does.

Having two different methods will just cause unnecessary confusion and extracting single number out of the count query might cause some unexpected behaviour when that is used for example in subqueries or otherwise as a part of more complex query (not sure if that is a real problem though).

However I would like to see count to be more compatible between databasese and it could have second option allowing to pass alias in which key count is stored `.count('', { as: 'count' })`.

Then later on even default for the count could be changed to be always `count` instead of DB specific value...

**Comment by kibertoad** [comment]
Released in 0.19.1
